src/pip/_internal/req/req_install.py

```diff
@@ -103,7 +103,7 @@ def __init__(
         global_options: Optional[List[str]] = None,
         hash_options: Optional[Dict[str, List[str]]] = None,
         constraint: bool = False,
-        extras: Iterable[str] = (),
+        extras: Collection[str] = (),
```

Comment:
`typing.Collection` is only available in 3.7+ so this needs to be stringified. Or you can keep using `Iterable` and change the `self.extras = extras` part fo first cast `extras` to a set.